addons/netfox.extras/state-machine/network-state.gd

```diff
@@ -0,0 +1,33 @@
+@icon("res://addons/netfox/icons/state-synchronizer.svg")
+extends Node
+class_name NetworkedState
+
+@export var state_machine: NetworkedStateMachine
+
+## A networked state for use with [NetworkedStateMachine].
+
+func _ready():
+	if state_machine == null and get_parent() is NetworkedStateMachine:
```

Comment:
We could make this a `@tool` script and move this validation logic to the editor. That way, users would see a little :warning: icon next to the node saying the state doesn't belong to any state machine.